cmd/serverconnect/main.go

```diff
@@ -228,6 +246,36 @@ func run(flags *flags) {
 	}
 }
 
+func newH1Server(f *flags, handler http.Handler) http.Server {
+	h1Server := http.Server{
+		Addr:    ":" + f.h1Port,
+		Handler: handler,
+	}
+	if !f.insecure {
+		h1Server.TLSConfig = newTLSConfig(f.certFile, f.keyFile)
+	}
+	return h1Server
+}
+
+func newH2Server(f *flags, handler http.Handler) http.Server {
+	h2Server := http.Server{
```

Comment:
Not sure if this should be `http2.Server`. The previous code uses `http`, so I kept it the same.

Reply:
Updated to use `h2c.NewHandler`. I tested this locally with an insecure client using `connect-h2` and `insecure` as the implementation and it worked. Also tested `connect-h1` and `insecure`, which also worked.